extensions/mongodb/src/main/java/com/hazelcast/jet/mongodb/impl/MongoConnection.java

```diff
@@ -111,5 +118,19 @@ public void close() {
             mongoClient.close();
             mongoClient = null;
         }
+        if (dataLink != null) {
+            dataLink.release();
```

Comment:
maybe assign dataLink to null too ?

Reply:
If connection is shared, then a client proxy is returned that closes the connection only after last connection being closed.